### DISCUSSION TITLE
One View - Multiple View Models

### DISCUSSION THREAD

**Discussion by sylvesp** [PrismLibrary/Prism#2779] [WPF]
Would it be possible to use THE SAME View with Multiple ViewModels?
I have a situation where the UI is exactly the same - a list of strings and a few buttons - but there are 3 different scenarios where the ViewModel has to behave differently when you click those buttons.
Also I would like to use this View-ViewModels as navigation targets in a region.

I've thought that there may be a way to register the SAME view 3 Times with Different ViewModels and have them named differently, but it looks that always the last registration wins.

Any ideas about would be the best way to deal with this situation?

Thanks,
Peter

**Comment by dansiegel** [comment]
This is currently possible in Prism.Maui and may or may not be available for other platforms in the future.
